common/app/controllers/EmailSignupController.scala

```diff
@@ -108,37 +128,74 @@ class EmailSignupController(
+        val identityNewsletter = EmailNewsletter.fromIdentityName(listName)
+        identityNewsletter match {
+          case Some(newsletter) =>
+            Cached(1.day)(
+              RevalidatableResult.Ok(
+                views.html.emailFragment(
+                  emailLandingPage,
+                  emailType,
+                  newsletter.identityName,
+                  newsletter.emailEmbed,
+                ),
+              ),
+            )
           case _ => Cached(15.minute)(WithoutRevalidationResult(NoContent))
         }
       }
     }
 
-  def subscriptionResult(result: String): Action[AnyContent] =
+  def subscriptionResult(result: String, listName: String): Action[AnyContent] =
     Action { implicit request =>
-      Cached(7.days)(result match {
-        case "success" => RevalidatableResult.Ok(views.html.emailSubscriptionResult(emailLandingPage, Subscribed))
-        case "invalid" => RevalidatableResult.Ok(views.html.emailSubscriptionResult(emailLandingPage, InvalidEmail))
-        case "error"   => RevalidatableResult.Ok(views.html.emailSubscriptionResult(emailLandingPage, OtherError))
-        case _         => WithoutRevalidationResult(NotFound)
-      })
-
+      listName match {
+        case "footer" => {
+          Cached(7.days)(result match {
+            case "success" =>
+              RevalidatableResult.Ok(views.html.emailSubscriptionResultFooter(emailLandingPage, Subscribed))
+            case "invalid" =>
+              RevalidatableResult.Ok(views.html.emailSubscriptionResultFooter(emailLandingPage, InvalidEmail))
+            case "error" =>
+              RevalidatableResult.Ok(views.html.emailSubscriptionResultFooter(emailLandingPage, OtherError))
+            case _ => WithoutRevalidationResult(NotFound)
+          })
+        }
+        case listName: String => {
+          val identityNewsletter = EmailNewsletter.fromIdentityName(listName)
+          identityNewsletter match {
+            case Some(newsletter) =>
+              Cached(7.days)(result match {
+                case "success" =>
+                  RevalidatableResult.Ok(
+                    views.html.emailSubscriptionResult(emailLandingPage, Subscribed, newsletter.emailEmbed),
+                  )
+                case "invalid" =>
+                  RevalidatableResult.Ok(
+                    views.html.emailSubscriptionResult(emailLandingPage, InvalidEmail, newsletter.emailEmbed),
+                  )
+                case "error" =>
+                  RevalidatableResult.Ok(
+                    views.html.emailSubscriptionResult(emailLandingPage, OtherError, newsletter.emailEmbed),
+                  )
+                case _ => WithoutRevalidationResult(NotFound)
+              })
+            case _ => Cached(15.minute)(WithoutRevalidationResult(NoContent))
+          }
+        }
+      }
     }
 
-  def submit(): Action[AnyContent] =
+  def submit(listName: String): Action[AnyContent] =
```

Comment:
I wonder whether we can pull `listName` out of the form and avoid changing this controller method's arguments and associated route.

Reply:
This was a great shout and I've implemented that now. I've also moved "success" and "non-success" states into separate endpoints/methods so that if the form fails to parse, we don't require the listName from it in order to generate the error state message